src/service/service.py

```diff
@@ -156,6 +177,18 @@ async def message_generator(
                 # So we only print non-empty content.
                 yield f"data: {json.dumps({'type': 'token', 'content': convert_message_content_to_string(content)})}\n\n"
             continue
+    
+    # Handle interruptions
+    snapshot = await agent.aget_state(kwargs["config"])
+    if snapshot.next: 
+        try:
+            ai_message = langchain_to_chat_message(snapshot.values["messages"][-1])
+            ichat_message = interrupt_from_call_tool(call_tool=ai_message.tool_calls[0])
```

Comment:
An AIMessage can include multiple tool calls. This would drop the additional ones. That seems bad.